### PR TITLE
fix: talosctl edit mc loop

### PR DESCRIPTION
**Pull request by frezbo**
Fixes re-opening editor forever when using `talosctl edit mc`.
Also fixes the temp dir getting filled up with temporary files created
for editing machine config.

Fixes: #6098

Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

**Comment by frezbo** [comment]
Code mostly borrowed from https://github.com/kubernetes/kubectl/blob/4da03973dd2fcd4645f20ac669d8a73cb017ff39/pkg/cmd/util/editor/editoptions.go#L315

File: cmd/talosctl/cmd/talos/edit.go [hunk 1]
```diff
@@ -83,11 +88,7 @@ func editFn(c *client.Client) func(context.Context, string, resource.Resource, e
 			}
 
 			if lastError != "" {
-				lines := strings.Split(lastError, "\n")
-
-				_, err = w.Write([]byte(
-					fmt.Sprintf("# \n# %s\n", strings.Join(lines, "\n# ")),
```
Comment (smira) [inline cmd/talosctl/cmd/talos/edit.go:89]: I think this had some `CRLF` magic for Windows due to the special writer being used which is not handled by the new function

Comment (frezbo) [reply cmd/talosctl/cmd/talos/edit.go:89]: it's the same code as the old one

Comment (frezbo) [reply cmd/talosctl/cmd/talos/edit.go:89]: https://github.com/siderolabs/talos/blob/e3c96117536a7dc55fb2f5817b1631819b7c6534/cmd/talosctl/cmd/talos/edit.go#L77 it's here

Comment (smira) [reply cmd/talosctl/cmd/talos/edit.go:89]: ah... I see, sorry, we're still using the `w.Write`... I somehow looks at the signature and ignored that

**Review by smira — APPROVED** [review]

**Pull request merged**